src/event-log/event-log-level.ts

```diff
@@ -15,101 +18,114 @@ type EventLogLevelConfig = {
+   */
+  async queryEvents(tenant: string, filters: Filter[], watermark?: string): Promise<Event[]> {
+    return await this.index.query(tenant, filters, { sortProperty: 'watermark', cursor: watermark });
+  }
 
-    return events;
+  /**
+   * Gets Events for a given tenant. An optional watermark is specified to get events beyond a certain point in time.
+   *
+   * @param tenant
+   * @param options options with that take an optional watermark.
+   * @returns an array of Events for the given tenant, if a watermark is provided only Events appended after tha watermark will be returned.
+   */
+  async getEvents(tenant: string, options?: GetEventsOptions): Promise<Event[]> {
+    return await this.index.query(tenant, [], { sortProperty: 'watermark', sortDirection: SortDirection.Ascending, cursor: options?.gt });
   }
 
-  async deleteEventsByCid(tenant: string, cids: Array<string>): Promise<number> {
-    if (cids.length === 0) {
+  /**
+   * Deletes Events associated with a given cid. Deletes any indexes that were created along with it as well.
+   * @param tenant
+   * @param messageCids an array of messageCids to delete.
+   * @returns the number of events deleted.
+   */
+  async deleteEventsByCid(tenant: string, messageCids: Array<string>): Promise<number> {
+    if (messageCids.length === 0) {
       return 0;
     }
 
-    const tenantEventLog = await this.db.partition(tenant);
-    const cidLog = await tenantEventLog.partition(CIDS_SUBLEVEL_NAME);
+    const cidLog = await this.messageCidPartition(tenant);
+    const cidOps: LevelWrapperBatchOperation<string>[] = [];
+    const indexDeletePromises: Promise<void>[] = [];
 
-    let ops: LevelWrapperBatchOperation<string>[] = [];
-    const promises: Array<Promise<string | undefined>> = [];
-
-    for (const cid of cids) {
-      ops.push({ type: 'del', key: cid });
-
-      const promise = cidLog.get(cid).catch(e => e);
-      promises.push(promise);
-    }
-
-    await cidLog.batch(ops);
-
-    ops = [];
     let numEventsDeleted = 0;
-
-    const watermarks: Array<string | undefined> = await Promise.all(promises);
-    for (const watermark of watermarks) {
-      if (watermark) {
-        ops.push({ type: 'del', key: watermark });
-        numEventsDeleted += 1;
+    for (const messageCid of messageCids) {
+      const watermark = await cidLog.get(messageCid);
+      if (watermark === undefined) {
+        continue;
       }
+      cidOps.push({ type: 'del', key: messageCid });
+      indexDeletePromises.push(this.index.delete(tenant, watermark));
+      numEventsDeleted += 1;
     }
 
-    const watermarkLog = await tenantEventLog.partition('watermarks');
-    await watermarkLog.batch(ops);
-
+    await cidLog.batch(cidOps);
+    await Promise.all(indexDeletePromises);
     return numEventsDeleted;
   }
+
+  /**
+   * Gets the LevelDB Sublevel for the messageCid reverse lookup.
+   */
+  private async messageCidPartition(tenant: string): Promise<LevelWrapper<string>> {
```

Comment:
Please rename.

```suggestion
  private async getMessageCidToWatermarkTable(tenant: string): Promise<LevelWrapper<string>> {
```

Of if you think too long:

```suggestion
  private async getWatermarkTable(tenant: string): Promise<LevelWrapper<string>> {
```